### DISCUSSION TITLE
SteamGuard-esque location-based 2FA?

### DISCUSSION THREAD

**Discussion by tyteen4a03** [ory/kratos#2183] [Q&A] [answered]
I am wondering if there's a plugin / feature that can enable something like SteamGuard where if the user logs in from an unfamiliar PC, the system sends them an email to confirm it's really them trying to access their account?

**Comment by vinckr** [comment, marked as answer]
Ory Kratos does have a concept of `privileged sessions`.
https://www.ory.sh/kratos/docs/concepts/session/#privileged-sessions
Maybe those could be extended for this functionality. 
There are also [Hooks](https://www.ory.sh/kratos/docs/next/self-service/hooks) for all kinds of custom logic, but making this a core feature is probably worth considering.
I don't think a current proposal/issue exists, but @zepatrik probably has some idea on this already 😉 .

**Reply by vinckr** [reply]
Hey @tyteen4a03,
Just checking in if your question was answered or if there was still something unclear.
I will mark this as answered for now, but feel free to come back to the discussion any time.

Thanks a ton! 🐝
